doc/contrib/coding_guide.rst

```diff
@@ -0,0 +1,158 @@
+################
+Coding Guideline
+################
+
+**Contents**
+
+.. contents::
+  :backlinks: none
+  :local:
+
+********************
+C++ Coding Guideline
+********************
+- Follow `Google style for C++ <https://google.github.io/styleguide/cppguide.html>`_.
```

Comment:
Are we following the 80-character line length? I see a lot of files with longer lines. Perhaps 100?

Reply:
We actually enforce 100-character line length instead: https://github.com/dmlc/dmlc-core/blob/f1ff6cc117f4e95169a9f62be549c8fe3e15c20f/scripts/lint.py#L72

Let me update the doc. I should also mention that we allow the use of C++ exception.